### PR TITLE
feat: support fish shell completion

### LINKED ISSUE #879: Add fish shell completion
## Describe the proposal
adding fish shell completion feature would be nice for fish shell users.

## Release Note

RELEASE NOTE:

### PR DESCRIPTION
**Pull request by alexlai97**
# Description

Add support for fish shell completion.

## Issue reference

closes #879 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by alexlai97** [comment]
Hello, I do know know if this feature has a unit test for it so i haven't updated unit test. And for documentation I think I updated everywhere I could find (only within this file). If there's other places I should update this documentation, plz let me know.

**Review by mukundansundar — DISMISSED** [review]

**Comment by mukundansundar** [comment]
LGTM. Thanks for the contribution.

File: cmd/completion.go [hunk 1]
```diff
@@ -112,6 +119,20 @@ func newCompletionZshCmd() *cobra.Command {
 	return cmd
 }
 
+func newCompletionFishCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "fish",
+		Short: "Generates fish completion functions",
```
Comment (shubham1172) [inline cmd/completion.go:125]:
```suggestion
		Short: "Generates fish completion scripts",
```

File: cmd/completion.go [hunk 2]
```diff
@@ -112,6 +119,20 @@ func newCompletionZshCmd() *cobra.Command {
...
+		Run: func(cmd *cobra.Command, args []string) {
+			RootCmd.GenFishCompletion(os.Stdout, true)
+		},
+	}
+
```
Comment (shubham1172) [inline cmd/completion.go:130]:
```suggestion
```

File: cmd/completion.go [hunk 3]
```diff
@@ -57,6 +57,12 @@ var completionExample = `
 	# Set the dapr completion code for zsh[1] to autoload on startup
   	dapr completion zsh > "${fpath[1]}/_dapr"
 
+	# Installing fish completion on Linux
```
Comment (shubham1172) [inline cmd/completion.go:60]: Can it be also used on macOS? Also is any `fish-completion` package required on either Linux or macOS?

File: cmd/completion.go [hunk 4]
```diff
@@ -57,6 +57,12 @@ var completionExample = `
...
+	## Load the dapr completion code for fish into the current shell
+	dapr completion fish | source
+	# Set the dapr completion code for fish[1] to autoload on startup
+	dapr completion fish > $HOME/.config/fish/functions/dapr.fish
```
Comment (shubham1172) [inline cmd/completion.go:64]: We can do similar to line 49, i.e. first create completion file, and then `source ~/<file>` it.

**Review by shubham1172 — CHANGES_REQUESTED** [review]
Thanks for the PR, just some nits and a question.

File: cmd/completion.go (hunk 3 above)
Comment (alexlai97) [reply cmd/completion.go:60]: Sorry I am not familiar with macOS or windows environment.

 `fish-completion` is not required, it's a built in feature.

**Comment by shubham1172** [comment]
The DCO is broken again @alexlai97

**Review by shubham1172 — APPROVED** [review]

**Pull request merged**